auxx/src/Command/DumpBlockchain.hs

```diff
@@ -164,3 +117,48 @@ dumpBlockchain outFolder = withStateLock HighPriority "auxx" $ \_ -> do
         maybeThrow (DBMalformed $ sformat errFmt hash) =<< getBlock hash
       where
         errFmt = "getBlockThrow: no block with HeaderHash: "%shortHashF
+
+----------------------------------------------------------------------------
+-- Loading a dump
+----------------------------------------------------------------------------
+
+-- | Load blocks from a file or directory and apply them.
+applyBlockchainDump
+    :: (MonadAuxxMode m, MonadIO m)
+    => FilePath -> m ()
+applyBlockchainDump path = withStateLock HighPriority "auxx" $ \_ -> do
+    isDir <- liftIO $ doesDirectoryExist path
+    if isDir then applyDir path else applyEpoch path
+
+-- | Apply all epochs from a directory.
+applyDir
```

Comment:
I don't like functions that take more than 1.5 screens, and usually my heuristic is “if I can write a Haddock docstring for it, it probably can be moved out of `where`”. It also helps that when the function is out of `where`, I can jump to its definition :)

Reply:
I feel that moving everything into a `where` clause is superior here because these functions are auxiliary with respect to `applyBlockchainDump` and aren't used anywhere else, while keeping them outside of `where` hints at the opposite. On the other hand, `where` introduces more visual clutter.

In this case, I'd probably weakly agree with you.